langchain4j/src/main/java/dev/langchain4j/service/DefaultAiServices.java

```diff
@@ -204,32 +207,23 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Exceptio
                             }
                         }
 
-                        Response<AiMessage> response;
+                        ChatRequest.Builder chatRequestBuilder = ChatRequest.builder()
+                                .messages(messages)
+                                .toolSpecifications(toolSpecifications);
+
                         if (supportsJsonSchema && jsonSchema.isPresent()) {
-                            ChatRequest chatRequest = ChatRequest.builder()
-                                    .messages(messages)
-                                    .toolSpecifications(toolSpecifications)
-                                    .responseFormat(ResponseFormat.builder()
-                                            .type(JSON)
-                                            .jsonSchema(jsonSchema.get())
-                                            .build())
+                            ResponseFormat responseFormat = ResponseFormat.builder()
+                                    .type(JSON)
+                                    .jsonSchema(jsonSchema.get())
                                     .build();
+                            chatRequestBuilder.responseFormat(responseFormat);
+                        }
 
-                            ChatResponse chatResponse = context.chatModel.chat(chatRequest);
+                        ChatRequest chatRequest = chatRequestBuilder.build();
 
-                            response = new Response<>(
-                                    chatResponse.aiMessage(),
-                                    chatResponse.tokenUsage(),
-                                    chatResponse.finishReason()
-                            );
-                        } else {
-                            // TODO migrate to new API
-                            response = toolSpecifications == null
-                                    ? context.chatModel.generate(messages)
-                                    : context.chatModel.generate(messages, toolSpecifications);
-                        }
+                        ChatResponse chatResponse = context.chatModel.chat(chatRequest);
 
-                        TokenUsage tokenUsageAccumulator = response.tokenUsage();
+                        TokenUsage tokenUsageAccumulator = ifNotNull(chatResponse.metadata(), ChatResponseMetadata::tokenUsage);
```

Comment:
This is really a nitpick, so feel free to ignore it, but this `ifNotNull` method seems to mock the groovy elvis operator and I'm not sure I like this. A better alternative imo would be to add a `tokenUsage` method directly to the `ChatResponse` object and perform the null check internally to that method.

A little bit more importantly it seems to me that the `ChatResponse`'s `metadata` can [never be null](https://github.com/langchain4j/langchain4j/pull/1989/files#diff-34c44ba239e7a80670ada445c8ad2ff7c5e99b7c592c27b9c00daa6dc6db37cdR21), so this check is totally useless here? I'm asking because maybe we may want to state more clearly what can or cannot be empty (null) in this API, and eventually use `Optional`s in these cases, so using the standard ifNotNull behavior that they provide.

Reply:
Oh good catch! I've added `ensureNotNull(builder.metadata, "metadata")` after `ifNotNull`, let me get rid of it

Reply:
Good point about being explicit about optionality, but this requires more analysis. Perhaps we should look into this after all/most other integrations implement this API so it will be clearer what is always present and what is optional

Reply:
When specific integration implements common tests (like `AbstractChatModelIT`, ) there will be some `assertXyz() { return false; }`, see OllamaChatModelIT as an example. Thist will help us understand which providers do not output specific parts of the response